subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitJupiterMultiVersionTest.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.junitplatform
+
+import org.gradle.integtests.fixtures.MultiVersionIntegrationSpec
+import org.gradle.testing.fixture.AbstractJUnitMultiVersionIntegrationTest
+
+trait JUnitJupiterMultiVersionTest {
+    AbstractJUnitMultiVersionIntegrationTest.BuildScriptConfiguration getBuildScriptConfiguration() {
+        return new JUnitJupiterBuildScriptConfiguration()
+    }
+
+    String getDependencyVersion() {
+        return JUnitJupiterBuildScriptConfiguration.getDependencyVersion()
+    }
+
+    static class JUnitJupiterBuildScriptConfiguration implements AbstractJUnitMultiVersionIntegrationTest.BuildScriptConfiguration {
+        String configureTestFramework = "useJUnitPlatform()"
+
+        @Override
+        String getTestFrameworkDependencies() {
+            return """
+                testImplementation 'org.junit.jupiter:junit-jupiter-api:${dependencyVersion}'
+                testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:${dependencyVersion}'
+                testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
+            """
+        }
+
+        static String getDependencyVersion() {
+            return MultiVersionIntegrationSpec.version.toString().substring("Jupiter:".length())
```

Comment:
Will this prefixing oddity go away after the refactoring is finished? It seems only necessary for `JUnitCoverage.JUNIT_VINTAGE_JUPITER` to work, and if we really need it I feel like there should be a better mechanism.

At the very least, if this will be kept, I'd like to assert that it starts with `Jupiter:` rather than blindly cutting it.

Reply:
Yeah, I'd like to.  I wanted to get as much under control as possible and then look and see what was left over in terms of stuff that is using these variables.  Hopefully, we can then remove these prefixes.